reactor/features/accounts/connect.py

```
@@ -42,10 +42,19 @@ def connect_account(account_details, cfg, client_details, reactor_account_id):
     logger.info(f'connect_account | Attempting to connect account {account_id}')
     logger.debug(f'connect_account | {account_details}')
     slack = SlackWeb(cfg.get('slack_webhook_url'))
+
+    # check if we have registered this account before, if so do not generate new env-id
+    env_dbh = dynamodb.CZReactorEnvironments()
+    try:
+        environment = env_dbh.get_from_account_id(account_id)
+        environment_id = environment['environment_id']
+    except dynamodb.DataNotFound:
```

I have mixed feelings about making a "safe" operation throw exceptions, but OTOH python loves throwing exceptions =)

---

Yes. I think we should reserve our own exceptions for truly out-of-the-ordinary cases. For example, if `get_from_account_id` returning `{}` is "normal", it should do so and this code can use `.get('environment_id')` instead. As a general rule of 👍, we may want to adopt a bit of Erlang's "let it crash" ... and only catch exceptions if the code can recover.

---

I don’t want it to crash, I want to catch the exception because the function has an lru cache wrapper. Any return value other than exception would be cached.

Also embrace exceptions and control flow, this is python

---

Good point about the LRU cache - that makes sense if you don't want to cache `None`